Start systemd symlink service in the role

jsf9k opened the pull request:
## 🗣 Description ##

This pull request adds tasks to this Ansible role that allow it to start the user-mode `systemd` service that creates a symlink to the COOL file share.

## 💭 Motivation and context ##

We want to enable this service for the VNC user in all our COOL AMIs, not [just in cisagov/pca-gophish-composition-packer](https://github.com/cisagov/pca-gophish-composition-packer/blob/develop/src/vnc.yml#L40-L95).

## 🧪 Testing ##

All `pre-commit` hooks and `molecule` tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] _All_ future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.

jsf9k committed:
Add tasks to start and enable the file share symlink service

This is done only for the users that are specified via a role
variable.

jsf9k committed:
Alphabetize Galaxy tags

jsf9k committed:
Remove paragraph that is now false

jsf9k committed:
Correct the role name in the usage example

jsf9k committed:
Move some tasks from the side effect playbook to the prepare playbook

We need the test user to be created earlier since we want to let the
role enable the user-mode systemd service.

Also remove some tasks that are now redundant since the tasks to start
and enable the systemd service were added to the role.

jsf9k committed:
Let the role enable the systemd service for the test user

Previously this was done by the side effect playbook.

jsf9k commented:
The Kali-based `molecule` testing will fail until cisagov/skeleton-ansible-role#92 is merged into this repository.  I'll rebase this branch and re-run the GitHub Actions once that happens.

jsf9k committed:
Add some tags to tasks to pass idempotence

Our trick of enabling linger, enabling the user-mode service, then
disabling linger causes some idempotence woe.  I don't see how to
check the status of the user-mode service cleanly, so I add some
molecule-idempotence-notest tags to two tasks to get around this.

dav3r reviewed (approved):
👍 👍

jsf9k committed:
Make the enabling of the systemd user-mode service work outside of Molecule

The task that actually enables the user-mode systemd service can
succeed when run via Molecule, but as far as I can tell the only way
to make this work is to set "become" to "no" and set
"ansible_ssh_user" to "{{ username }}".  Unfortunately this
configuration cannot work when this role is used to build an AMI,
since in that case ssh connects via an AWS-generated ssh key that is
only valid for the AMI's default user.  Therefore we take the even
more drastic measure of applying the molecule-notest tag to this task.

I created #29 as a reminder to figure out a way to make the
ansible.builtin.systemd task succeed when run via molecule or when
this Ansible role is used to built an AMI.

mcdonnnj commented on molecule/default/prepare.yml:
```diff
@@ -31,3 +31,21 @@
+
+# This is necessary to setup for installing this Ansible role with the
+# specification that the test user have a user-mode systemd service
+# that creates a symlink to the /share/directory
```
```suggestion
# that creates a symlink to the /share directory.
```

mcdonnnj reviewed (approved):
Strong work 💪💪💪. I have one comment tweak for your consideration.

jsf9k committed:
Fix typo

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove playbook that is no longer necessary

The user-mode systemd service is now started in the Ansible role
itself, so there is no longer a need to do this in the side effect
playbook.

jsf9k committed:
Comment out a test that cannot pass

The test can be uncommented once #29 is resolved.

dav3r reviewed (approved):
Still excellent! 👍
